uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js

```diff
@@ -173,7 +173,7 @@ define([
       };
 
       $scope.cancel = function () {
-        if ($scope.documentForm.$pristine) {
+        if ($scope.documentForm.$pristine && angular.equals(files, $scope.files) && !$scope.uploader.queue.length) {
```

Comment:
Currently,  the confirmation popup is only displayed if not fields data (except file attachment changes) in fields are changed, and when he tries to cancel the form. So, i have added two more conditions:
1. `angular.equals(files, $scope.files)` which is to check if the files  have been changed
2.`!$scope.uploader.queue.length` which is to check if there are files in the  uploader queue.
I am displaying the confirmation, also for when there is change in files and when there are files in the uploader queue and if user tries to cancel the form.